### PR TITLE
Tag path supprt

### PR DESCRIPTION
**Pull request by sukolenvo**

### CODE REVIEW

File: src/wt/api/wt_common.proto [hunk 1]
```diff
@@ -41,14 +41,32 @@ message CreateTagRequest {
...
     // displayed in the GUI and search functions
+    // ignored if tag_type is FOLDER
     string tag_description = 3;
 
     // optional: if modifier should be attached by default for this tag
     // such as as `billable` or `research` for use in integration
-    string default_modifier = 4;
+    string default_modifier = 4 [deprecated = true];
+
+    TagType tag_type = 5;
```
Comment (vyshane) [inline src/wt/api/wt_common.proto]: If we provide a `tag_path` that doesn't currently exist, will a folder be automatically created to contain the new tag? If that's the case, I can't think of a use case for the agent to create empty folders, and we can omit `tag_type`.

Comment (sukolenvo) [reply src/wt/api/wt_common.proto]: Yes, folder is created automatically

Comment (vyshane) [reply src/wt/api/wt_common.proto]: Okay, let's omit `tag_type` and always only create tags. The agent then doesn't need to know about the concept of tag folders.

**Comment by vyshane** [comment]
Thanks, merged

**Pull request merged**